packages/android_intent/android/src/main/java/dev/flutter/plugins/androidintent/AndroidIntentPlugin.java

```diff
@@ -0,0 +1,66 @@
+package dev.flutter.plugins.androidintent;
+
+import androidx.annotation.NonNull;
+import io.flutter.embedding.engine.plugins.FlutterPlugin;
+import io.flutter.embedding.engine.plugins.activity.ActivityAware;
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.PluginRegistry.Registrar;
+
+/**
+ * The new embedding implementation of the plugin.
+ *
+ * <p>This can be included in an add to app scenario to gracefully handle activity and context
+ * changes, unlike the previous {@link io.flutter.plugins.androidintent.AndroidIntentPlugin}.
+ */
+public class AndroidIntentPlugin implements FlutterPlugin, ActivityAware {
+  private final IntentSender sender;
+
+  /**
+   * Initialize this within the {@code #configureFlutterEngine} of a Flutter activity or fragment.
+   *
+   * <p>See {@code dev.flutter.plugins.androidintentexample.MainActivity} for an example.
+   */
+  public AndroidIntentPlugin() {
+    sender = new IntentSender(/*activity=*/ null, /*context=*/ null);
+  }
+
+  /** This exists for legacy compatibility purposes. */
+  public static void registerWith(Registrar registrar) {
```

Comment:
Doesn't seem to be called in this PR.
We might not need this. The io MainAcitivity can be left unchanged.

Reply:
Ah, right. I was thinking there was some magic here but since I didn't delete the `io.` class this isn't necessary. Deleted.